android/app/src/main/java/com/reactnativenavigation/views/BottomTabs.java

```diff
@@ -132,23 +127,52 @@ private void createVisibilityAnimator() {
                 Constants.BOTTOM_TABS_HEIGHT);
     }
 
-    private void setStyle() {
-        if (hasBadgeBackgroundColor()) {
+    private void setStyle(StyleParams params) {
+        if (hasBadgeBackgroundColor(params)) {
             setNotificationBackgroundColor(AppStyle.appStyle.bottomTabBadgeBackgroundColor.getColor());
         }
-        if (hasBadgeTextColor()) {
+        if (hasBadgeTextColor(params)) {
             setNotificationTextColor(AppStyle.appStyle.bottomTabBadgeTextColor.getColor());
         }
+
+        if (hasBottomTabsColor(params)) {
+            setBackgroundColor(params.bottomTabsColor);
+        }
+        if (hasBottomTabsButtonColor(params)) {
+            if (getInactiveColor() != params.bottomTabsButtonColor.getColor()) {
+                setInactiveColor(params.bottomTabsButtonColor.getColor());
+            }
+        }
+        if (hasSelectedBottomTabsButtonColor(params)) {
+            if (getAccentColor() != params.selectedBottomTabsButtonColor.getColor()) {
+                setAccentColor(params.selectedBottomTabsButtonColor.getColor());
+            }
+        }
+    }
+
+    private boolean hasBadgeTextColor(StyleParams params) {
+        return params.bottomTabBadgeTextColor != null &&
```

Comment:
We can avoid adding these convince method by initializing these style properties so they are never `null` [here](https://github.com/wix/react-native-navigation/blob/master/android/app/src/main/java/com/reactnativenavigation/params/StyleParams.java#L143). The `Color` class is meant to be used as a null object

Reply:
Of cause. This was original coding, so I've adapted it to match this code style.
Which color will be the default one?

Reply:
No need to specify a default color since in this case `hasColor()` will return false as expected.

Reply:
so you will only check `hasColor()` in the `setStyle` function and remove the `has...`functions?

Reply:
Yes exactly :)